tests/unit/value_handler.cc

```diff
@@ -0,0 +1,11 @@
+#include "../googletest/include/gtest/gtest.h"
+#include "../../src/value.h"
+#include "test_value.c"
```

Comment:
Including c files seems weird to me,
can't test_value use the google test framework?
or can't we just have a test_value.h and include that?

I'm not sure of the purpose of value_handler.cc, I though we would re-write test_value to use the GoogleTest framework.

Reply:
We don't need to include c files, that can be avoided.

My concern (which is perhaps unfounded) is that to directly use the google test framework, we need to compile our input test files as C++. This requires minor changes to match C++ conventions, like explicitly casting all `malloc` returns - nothing too painful.
I worry, however, that building our test suite so differently from our actual code base is inevitably going to diminish the relevance of our testing.

By compiling our test files as C with gcc/clang, then linking their object files with a separate C++ handler that contains all the google test logic, I feel like we'll more accurately test the code base as it will be invoked by a user.

This is kind of conjectural for me, though, so I welcome other opinions! @mnunberg , it would be great to hear your thinking on this as well.